template/contracts/NFTCollection.sol

```diff
@@ -0,0 +1,71 @@
+//Contract based on [https://docs.openzeppelin.com/contracts/3.x/erc721](https://docs.openzeppelin.com/contracts/3.x/erc721)
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+
+contract NFTCollection is ERC721URIStorage, Ownable {
+    uint256 public counter = 0;
+
+    mapping(uint256 => Nft) public allNfts;
+    mapping(string => bool) public tokenURIExists;
+
+    struct Nft {
+        uint256 tokenId;
+        string tokenName;
+        string tokenURI;
+        address payable currentOwner;
+        uint256 price;
+    }
+
+    constructor() public ERC721("NFTCollection", "NFT") {}
+
+    function mintNft(
+        string memory tokenURI,
+        string memory tokenName,
+        uint256 price
+    ) public onlyOwner returns (uint256) {
+        counter++;
+
+        require(!tokenURIExists[tokenURI]);
+
+        _mint(msg.sender, counter);
+        _setTokenURI(counter, tokenURI);
+
+        address payable addr = payable(msg.sender);
+
+        Nft memory newNft = Nft(counter, tokenName, tokenURI, addr, price);
+
+        allNfts[counter] = newNft;
+
+        tokenURIExists[tokenURI] = true;
+    }
+
+    function buyNft(uint256 _tokenId) public payable {
+        require(msg.sender != address(0));
```

Comment:
Is this check required? Don't think msg.sender can ever be zero address, generally we check this for "to" addresses

Reply:
Removed in df5ad20